test/redular.js

```diff
@@ -195,31 +199,75 @@ describe('Redular', function () {
 
         setTimeout(() => {
             done();
-        }, 4000);
+        }, 3000);
     });
 
-    it('should be able to prune all data without a matching event', async function (done) {
+    it('should be able to prune all data without a matching event', async function () {
         Redular1.redis.set('redular-data:pruneTest', JSON.stringify({ valid: false }));
         await Redular1.pruneData();
         let data = await Redular1.redis.get('redular-data:pruneTest', function (error, data) {
             if (data) {
                 throw new Error('Data should have been pruned!');
             }
-            done();
         });
     });
 
-    it('should not prune data with matching event keys', async function (done) {
+    it('should not prune data with matching event keys', async function () {
         Redular1.defineHandler('pruneTestEvent', function (data) {
-            if (data.valid) {
-                done();
-            } else {
+            if (!data.valid) {
                 throw new Error('Incorrect handler called');
             }
         });
 
         var now = new Date();
-        Redular1.scheduleEvent('pruneTestEvent', now.setSeconds(now.getSeconds() + 3), false, { valid: true }, 'tester');
+        let eventId = Redular1.scheduleEvent('pruneTestEvent', now.setSeconds(now.getSeconds() + 1), false, { valid: true }, 'tester');
         await Redular1.pruneData();
+        Redular1.deleteEvent(eventId.event);
+    });
+
+    it('should be able to retrieve events that are within a date range', async function () {
+        let validEventIds = [];
+        var startDate = new Date();
+
+        var now = new Date();
+        validEventIds.push(Redular1.scheduleEvent('validDateRangeTestEvent', now.setSeconds(now.getSeconds() + 1), false, { valid: true }).event);
+        validEventIds.push(Redular1.scheduleEvent('validDateRangeTestEvent', now.setSeconds(now.getSeconds() + 1), false, { valid: true }).event);
+
+        var endDate = new Date();
+        endDate.setSeconds(endDate.getSeconds() + 2);
+
+        let eventsInRange = await Redular1.getEvents(startDate, endDate);
+        expect(eventsInRange).to.be.an('array');
+        validEventIds.sort();
+        eventsInRange.sort();
+
+        expect(eventsInRange).to.have.deep.members(validEventIds);
+        // Expire long events from this test
+        setTimeout(() => {}, 4);
```

Comment:
The timeout is in milliseconds, should be `4000` if you want to wait 4 sec. Does this need to happen btw? In your `afterEach` block, is the clear handlers not taking care of this?

Reply:
Clear handlers only removes the callback that gets called on event expiration.

Events need to be deleted still